paper-api/src/main/java/org/bukkit/structure/PlacementOptions.java

```diff
@@ -0,0 +1,266 @@
+package org.bukkit.structure;
+
+import com.google.common.base.Preconditions;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Random;
+import org.bukkit.block.structure.Mirror;
+import org.bukkit.block.structure.StructureRotation;
+import org.bukkit.util.BlockTransformer;
+import org.bukkit.util.EntityTransformer;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents options for placing a {@link Structure}.
+ */
+public class PlacementOptions {
+
+    private boolean includeEntities = true;
+    private StructureRotation structureRotation = StructureRotation.NONE;
+    private Mirror mirror = Mirror.NONE;
+    private int palette = -1;
+    private float integrity = 1;
+    private final Random random;
+    private boolean strict = false;
+    private boolean applyWaterlogging = true;
+    private Collection<BlockTransformer> blockTransformers = Collections.emptyList();
+    private Collection<EntityTransformer> entityTransformers = Collections.emptyList();
+
+    /**
+     * Create a new set of placement options for placing a {@link Structure}.
+     *
+     * @param random The randomizer used for setting the structure's
+     *               {@link org.bukkit.loot.LootTable LootTables} and integrity.
+     */
+    public PlacementOptions(@NotNull Random random) {
+        this.random = random;
+    }
+
+    /**
+     * Set whether to include entities when the structure is placed.
+     * <p>Will default to true if not set.</p>
+     *
+     * @param includeEntities Whether to include entities.
+     * @return This PlacementOptions.
+     */
+    public @NotNull PlacementOptions includeEntities(boolean includeEntities) {
+        this.includeEntities = includeEntities;
+        return this;
+    }
+
+    /**
+     * Whether to include entities when the structure is placed.
+     *
+     * @return Whether to include entities.
+     */
+    public boolean includeEntities() {
+        return includeEntities;
+    }
+
+    /**
+     * Set the rotation a structure will place with.
+     * <p>Will default to {@link StructureRotation#NONE} if not set.</p>
+     *
+     * @param structureRotation Rotation to place with.
+     * @return This PlacementOptions.
+     */
+    public @NotNull PlacementOptions structureRotation(@NotNull StructureRotation structureRotation) {
```

Comment:
```suggestion
     * @return This PlacementOptions.
     */
    @Contract(value = "_ -> this", mutates = "this")
    public @NotNull PlacementOptions structureRotation(@NotNull StructureRotation structureRotation) {
```

same for all setters

Reply:
I'll admit, I have never used contracts before, and frankly I have no clue how they work 😬 
I have taken what you have suggested and applied to setters, I hope I did this right, please confirm.

Reply:
yup that looks good, This exact contract just means:
For any argument passed ( _ ) this method will return ( -> ) the current object ( this )

Reply:
you forgot includeEntities

Reply:
It appears I did! Oops. Fixed!